async.zsh

```diff
@@ -279,6 +279,8 @@ _async_zle_watcher() {
 # 	async_job <worker_name> <my_function> [<function_params>]
 #
 async_job() {
+	# Reset all options to defaults inside async job.
+	emulate -R zsh
```

Comment:
We should change this to `emulate -L zsh` and remove the `setopt` below (`-L` does the same, and more). Using `-R` here would modify the options in the users shell since `LOCAL_OPTIONS` has not yet been set.